pilot/pkg/model/sidecar.go

```diff
@@ -711,25 +726,33 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 func matchingService(importedHosts hostClassification, service *Service, ilw *IstioEgressListenerWrapper) *Service {
 	matchPort := needsPortMatch(ilw)
 
-	// first, check exactHosts
-	if importedHosts.exactHosts.Contains(service.Hostname) {
+	if importedHosts.Matches(service.Hostname) {
 		if matchPort {
 			return serviceMatchingListenerPort(service, ilw)
 		}
 		return service
 	}
+	return nil
+}
 
-	// exactHosts not found, fallback to loop allHosts
-	for _, importedHost := range importedHosts.allHosts {
-		// Check if the hostnames match per usual hostname matching rules
-		if service.Hostname.SubsetOf(importedHost) {
-			if matchPort {
-				return serviceMatchingListenerPort(service, ilw)
-			}
-			return service
+// Return the original service or a trimmed service which has a subset of the ports in original service.
+func matchingAliasService(importedHosts hostClassification, service *Service) *Service {
+	if service == nil {
+		return nil
+	}
+	matched := make([]NamespacedHostname, 0, len(service.Attributes.Aliases))
+	for _, alias := range service.Attributes.Aliases {
+		if importedHosts.Matches(alias.Hostname) {
+			matched = append(matched, alias)
 		}
 	}
-	return nil
+
+	if len(matched) == len(service.Attributes.Aliases) {
+		return service
+	}
+	service = service.DeepCopy()
+	service.Attributes.Aliases = matched
```

Comment:
can you exlplain why also set aliases here

Reply:
added comment: `matchingAliasService the original service or a trimmed service which has a subset of aliases, based on imports from sidecar`

Its just like `matchingService`.